### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by array-in-a-matrix**
Adds support to run drainwine inside a docker container either using docker alone or using docker compose.

### CODE REVIEW

**Comment by kuroppoi** [comment]
Sorry, I changed some parts of the build logic. The output has been moved from `/build/libs` to `/build/dist`

**Comment by array-in-a-matrix** [comment]
Fixed, my fault for building on master branch instead of develop. Is there a need to pass the `disablegui` flag? or is that handled if no GUI is detected? As I've ran brainwine 1.0.3.1 with and without it on my server with no noticeable difference.

**Comment by kuroppoi** [comment]
It will automatically skip GUI creation if the platform doesn't support the Java Desktop API unless the `forcegui` flag is present.
So in this case it's _probably_ fine not to pass the `disablegui` flag, but it doesn't hurt if you do pass it, either.

File: Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ ARG GATEWAY_PORT=5001
 ARG SERVER_PORT=5002
 ARG PORTAL_PORT=5003
 EXPOSE $GATEWAY_PORT $SERVER_PORT $PORTAL_PORT
-CMD ["sh", "-c", "java -jar /app/brainwine.jar"]
+CMD ["sh", "-c", "java -jar /app/brainwine.jar --disablegui"]
```
Comment (kuroppoi) [inline Dockerfile]: Should just be `disablegui` instead of `--disablegui`

**Comment by array-in-a-matrix** [comment]
@kuroppoi  any other concerns?

**Comment by kuroppoi** [comment]
I don't think so, but I do have a question.
Is there any reason to use `docker run` instead of `docker compose up`? (I've never used Docker prior to this)

**Comment by array-in-a-matrix** [comment]
> I don't think so, but I do have a question. Is there any reason to use `docker run` instead of `docker compose up`? (I've never used Docker prior to this)

If someone wants to run brainwine alongside other docker containers or multiple instances, they may find configuring a compose file neater instead of running several run commands. For running brainwine only, there is no reason to choice one over the other. I prefer to debug using docker run and proper deployment using compose.

**Comment by kuroppoi** [comment]
Alright, just one more thing.
`docker run` is causing issues for me on Windows because of `$PWD`.
Changing the command to the following fixes it, but I'm not sure if it causes problems on other platforms:
```sh
docker run --name "brainwine" --volume /$PWD:/data brainwine:latest
```

I'd also like to put the setup instructions in its own section rather than under `Building`, if that's alright.

**Comment by array-in-a-matrix** [comment]
> Alright, just one more thing. `docker run` is causing issues for me on Windows because of `$PWD`. Changing the command to the following fixes it, but I'm not sure if it causes problems on other platforms:
> 
> ```shell
> docker run --name "brainwine" --volume /$PWD:/data brainwine:latest
> ```
> 
> I'd also like to put the setup instructions in its own section rather than under `Building`, if that's alright.

`/$PWD` seems to work fine but `${PWD}` and `$(pwd)` seem to be the proper [cross platform options](https://stackoverflow.com/questions/41485217/mount-current-directory-as-a-volume-in-docker-on-windows-10).

I moved the setup instructions to its own server setup section.

**Comment by array-in-a-matrix** [comment]
I've added `jar.Dockerfile`, this file can just build the `brainwine.jar` without setting up a server, so no need to install Java or other dependencies on host. `Dockerfile` file builds the project and sets up a server.

**Comment by kuroppoi** [comment]
It seems a bit overkill to build it this way, and running the jar requires Java anyway.

**Comment by kuroppoi** [comment]
I've changed/simplified the setup instructions a bit (sorry I'm very picky! 🤣)
Does it look okay like this or should anything be added?

**Comment by array-in-a-matrix** [comment]
No worries, looks good to me! :shipit:

**Comment by kuroppoi** [comment]
Awesome, thanks!

**Pull request merged**